src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,220 @@
+                                DateTimeFormatter formatter2 = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                                LocalDate parsedFromDate = LocalDate.parse(from, formatter2);
+                                LocalDateTime parsedToDateTime = LocalDateTime.parse(to, formatter1);
+                                Task newTask = new Event(content, isDone, parsedFromDate, parsedToDateTime);
+                                items.add(newTask);
+                            } else {
+                                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                                LocalDate parsedFromDate = LocalDate.parse(from, formatter);
+                                LocalDate parsedToDate = LocalDate.parse(to, formatter);
+                                Task newTask = new Event(content, isDone, parsedFromDate, parsedToDate);
+                                items.add(newTask);
+                            }
+                        } catch (Exception e) {
+                            throw new DukeException("event");
+                        }
+                        break;
+                    }
+                    default: {
+                        items.add(new Todo(splited[1], splited[2].equals("Y")));
+                    }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new DukeException("Error accessing / creating the file: " + e.getMessage());
+        }
+
+        return items;
+    }
+
+    /**
+     * Store tasks to the specified file path.
+     *
+     * @param tasks A list of existing tasks.
+     * @throws IllegalArgumentException If error in accessing file to store tasks.
+     */
+    public void saveChanges(TaskList tasks) throws DukeException{
+        String home = System.getProperty("user.dir");
+        Path filePath = Paths.get(home, this.filePath);
+
+        StringBuilder textToAdd = new StringBuilder();
+        for (Task t: tasks.getItems()) {
+            switch (t.getClass().getName()) {
+                case "duke.task.Deadline": {
+                    textToAdd.append(t.getClass().getName() + "`" + t.getFields()[0] + "`" + t.getFields()[1]
+                            + "`" + t.getFields()[2] + "/end");
+                    break;
+                }
+                case "duke.task.Event": {
+                    textToAdd.append(t.getClass().getName() + "`" + t.getFields()[0] + "`" + t.getFields()[1]
+                            + "`" + t.getFields()[2] + "`" + t.getFields()[3] + "/end");
+                    break;
+                }
+                default: {
+                    textToAdd.append(t.getClass().getName() + "`" + t.getFields()[0] + "`" + t.getFields()[1] + "/end");
+                }
+            }
+        }
+
+        try {
+            FileWriter fw = new FileWriter(filePath.toString());
+            fw.write(textToAdd.toString());
+            fw.close();
```

Comment:
I think you should close fw in a finally block. Because this would not close in case an exception is thrown

Reply:
That is a good catch, will fix it later, thanks!